drivers/hwinfo/hwinfo_nrf.c

```diff
@@ -76,16 +76,35 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 	if (reason & NRFX_RESET_REASON_DOG_MASK) {
 		flags |= RESET_WATCHDOG;
 	}
-	if (reason & NRFX_RESET_REASON_LOCKUP_MASK) {
+
+#if defined(NRF_RESETINFO)
+	if (NRFX_RESET_REASON_LOCAL_DOG0_MASK) {
+		flags |= RESET_WATCHDOG;
+	}
+#endif
+
+#if defined(NRF_RESETINFO)
+	if (reason & NRFX_RESET_REASON_LOCAL_LOCKUP_MASK)
+#else
+	if (reason & NRFX_RESET_REASON_LOCKUP_MASK)
```

Comment:
nRF54H20 has local and global lockup reset reasons, so shouldn't it be like this:
```
	if ((reason & NRFX_RESET_REASON_LOCKUP_MASK)
#if defined(NRF_RESETINFO)
	    || (reason & NRFX_RESET_REASON_LOCAL_LOCKUP_MASK)
#endif
        ) {
		flags |= RESET_CPU_LOCKUP;
        }
```
?